pages/queueListing.js

```diff
@@ -0,0 +1,112 @@
+import React, { useState, useEffect } from "react";
+import Header from "../components/Header";
+import Footer from "../components/Footer";
+import ShowList from "../components/ShowList";
+import { withRouter } from "next/router";
+
+const queueListing = props => {
+  const columns = React.useMemo(() => [
+    {
+      Header: "Job Id",
+      accessor: "JobId"
+    },
+    {
+      Header: "Title",
+      accessor: "Title"
+    }
+  ]);
+
+  const [state, setState] = useState({
+    waiting: [],
+    active: [],
+    delayed: [],
+    completed: [],
+    failed: [],
+    currentState: "waiting"
+  });
+  const fetchData = () => {
```

Comment:
Instead of calling fetchData here, use getServerSideProps. Check https://github.com/coderwassananmol/BUB2/blob/develop/pages/queue.js#L15